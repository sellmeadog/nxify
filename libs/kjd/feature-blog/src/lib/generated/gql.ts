/* eslint-disable */
import * as types from './graphql';
import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';

/**
 * Map of all GraphQL operations in the project.
 *
 * This map has several performance disadvantages:
 * 1. It is not tree-shakeable, so it will include all operations in the project.
 * 2. It is not minifiable, so the string of a GraphQL query will be multiple times inside the bundle.
 * 3. It does not support dead code elimination, so it will add unused operations.
 *
 * Therefore it is highly recommended to use the babel or swc plugin for production.
 */
const documents = {
    "\n  query ArticleQuery($slug: String!) {\n    ...ArticleContentQueryFragment\n  }\n": types.ArticleQueryDocument,
    "\n  query ArticleMetadataQuery($slug: String!) {\n    article(where: { slug: $slug }) {\n      author {\n        name\n      }\n      createdAt\n      hero {\n        caption\n        title\n      }\n      tags {\n        tag\n      }\n    }\n  }\n": types.ArticleMetadataQueryDocument,
    "\n  query PageQuery($slug: String) {\n    ...PageContentQueryFragment\n  }\n": types.PageQueryDocument,
    "\n  query PageMetadataQuery($slug: String) {\n    page(where: { slug: $slug }) {\n      createdAt\n      hero {\n        caption\n        title\n      }\n    }\n  }\n": types.PageMetadataQueryDocument,
    "fragment ArticleMetadata on Article {\n  author {\n    name\n  }\n  createdAt\n  excerpt\n  tags {\n    tag\n  }\n  title\n}\n\nquery ArticleMetadataBySlug($slug: String) {\n  article(where: {slug: $slug}) {\n    ...ArticleMetadata\n  }\n}": types.ArticleMetadataFragmentDoc,
    "\n  fragment ArticleAuthorFragment on Article {\n    author {\n      name\n      avatar {\n        url\n      }\n      biography\n    }\n  }\n": types.ArticleAuthorFragmentFragmentDoc,
    "\n  fragment ArticleContentQueryFragment on Query {\n    article(where: { slug: $slug }) {\n      ...ArticleHeroFragment\n      ...ArticleMarkdownFragment\n      ...ArticleAuthorFragment\n    }\n  }\n": types.ArticleContentQueryFragmentFragmentDoc,
    "\n  fragment ArticleExplorerSectionFragment on ArticleExplorer {\n    first\n    id\n    subtitle\n    title\n  }\n": types.ArticleExplorerSectionFragmentFragmentDoc,
    "\n  query ArticleExplorerQuery($first: Int) {\n    articlesConnection(first: $first, orderBy: createdAt_DESC) {\n      edges {\n        cursor\n        node {\n          ...ArticlePreviewFragment\n        }\n      }\n      pageInfo {\n        endCursor\n        hasNextPage\n        hasPreviousPage\n        pageSize\n        startCursor\n      }\n    }\n  }\n": types.ArticleExplorerQueryDocument,
    "\n  fragment ArticleHeroFragment on Article {\n    createdAt\n    hero {\n      caption\n      image {\n        url\n      }\n      subtitle\n      title\n    }\n  }\n": types.ArticleHeroFragmentFragmentDoc,
    "\n  fragment ArticleMarkdownFragment on Article {\n    markdown\n  }\n": types.ArticleMarkdownFragmentFragmentDoc,
    "\n  fragment ArticlePreviewFragment on Article {\n    author {\n      name\n    }\n    createdAt\n    hero {\n      caption\n      title\n      subtitle\n    }\n    id\n    slug\n  }\n": types.ArticlePreviewFragmentFragmentDoc,
    "\n  fragment PageContentQueryFragment on Query {\n    page(where: { slug: $slug }) {\n      ...PageHeroFragment\n      ...PageSectionExplorerFragment\n    }\n  }\n": types.PageContentQueryFragmentFragmentDoc,
    "\n  fragment PageHeroFragment on Page {\n    hero {\n      caption\n      image {\n        url\n      }\n      subtitle\n      title\n    }\n  }\n": types.PageHeroFragmentFragmentDoc,
    "\n  fragment PageSectionExplorerFragment on Page {\n    sections {\n      __typename\n      ...ArticleExplorerSectionFragment\n    }\n  }\n": types.PageSectionExplorerFragmentFragmentDoc,
};

/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 *
 *
 * @example
 * ```ts
 * const query = graphql(`query GetUser($id: ID!) { user(id: $id) { name } }`);
 * ```
 *
 * The query argument is unknown!
 * Please regenerate the types.
 */
export function graphql(source: string): unknown;

/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  query ArticleQuery($slug: String!) {\n    ...ArticleContentQueryFragment\n  }\n"): (typeof documents)["\n  query ArticleQuery($slug: String!) {\n    ...ArticleContentQueryFragment\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  query ArticleMetadataQuery($slug: String!) {\n    article(where: { slug: $slug }) {\n      author {\n        name\n      }\n      createdAt\n      hero {\n        caption\n        title\n      }\n      tags {\n        tag\n      }\n    }\n  }\n"): (typeof documents)["\n  query ArticleMetadataQuery($slug: String!) {\n    article(where: { slug: $slug }) {\n      author {\n        name\n      }\n      createdAt\n      hero {\n        caption\n        title\n      }\n      tags {\n        tag\n      }\n    }\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  query PageQuery($slug: String) {\n    ...PageContentQueryFragment\n  }\n"): (typeof documents)["\n  query PageQuery($slug: String) {\n    ...PageContentQueryFragment\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  query PageMetadataQuery($slug: String) {\n    page(where: { slug: $slug }) {\n      createdAt\n      hero {\n        caption\n        title\n      }\n    }\n  }\n"): (typeof documents)["\n  query PageMetadataQuery($slug: String) {\n    page(where: { slug: $slug }) {\n      createdAt\n      hero {\n        caption\n        title\n      }\n    }\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "fragment ArticleMetadata on Article {\n  author {\n    name\n  }\n  createdAt\n  excerpt\n  tags {\n    tag\n  }\n  title\n}\n\nquery ArticleMetadataBySlug($slug: String) {\n  article(where: {slug: $slug}) {\n    ...ArticleMetadata\n  }\n}"): (typeof documents)["fragment ArticleMetadata on Article {\n  author {\n    name\n  }\n  createdAt\n  excerpt\n  tags {\n    tag\n  }\n  title\n}\n\nquery ArticleMetadataBySlug($slug: String) {\n  article(where: {slug: $slug}) {\n    ...ArticleMetadata\n  }\n}"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  fragment ArticleAuthorFragment on Article {\n    author {\n      name\n      avatar {\n        url\n      }\n      biography\n    }\n  }\n"): (typeof documents)["\n  fragment ArticleAuthorFragment on Article {\n    author {\n      name\n      avatar {\n        url\n      }\n      biography\n    }\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  fragment ArticleContentQueryFragment on Query {\n    article(where: { slug: $slug }) {\n      ...ArticleHeroFragment\n      ...ArticleMarkdownFragment\n      ...ArticleAuthorFragment\n    }\n  }\n"): (typeof documents)["\n  fragment ArticleContentQueryFragment on Query {\n    article(where: { slug: $slug }) {\n      ...ArticleHeroFragment\n      ...ArticleMarkdownFragment\n      ...ArticleAuthorFragment\n    }\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  fragment ArticleExplorerSectionFragment on ArticleExplorer {\n    first\n    id\n    subtitle\n    title\n  }\n"): (typeof documents)["\n  fragment ArticleExplorerSectionFragment on ArticleExplorer {\n    first\n    id\n    subtitle\n    title\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  query ArticleExplorerQuery($first: Int) {\n    articlesConnection(first: $first, orderBy: createdAt_DESC) {\n      edges {\n        cursor\n        node {\n          ...ArticlePreviewFragment\n        }\n      }\n      pageInfo {\n        endCursor\n        hasNextPage\n        hasPreviousPage\n        pageSize\n        startCursor\n      }\n    }\n  }\n"): (typeof documents)["\n  query ArticleExplorerQuery($first: Int) {\n    articlesConnection(first: $first, orderBy: createdAt_DESC) {\n      edges {\n        cursor\n        node {\n          ...ArticlePreviewFragment\n        }\n      }\n      pageInfo {\n        endCursor\n        hasNextPage\n        hasPreviousPage\n        pageSize\n        startCursor\n      }\n    }\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  fragment ArticleHeroFragment on Article {\n    createdAt\n    hero {\n      caption\n      image {\n        url\n      }\n      subtitle\n      title\n    }\n  }\n"): (typeof documents)["\n  fragment ArticleHeroFragment on Article {\n    createdAt\n    hero {\n      caption\n      image {\n        url\n      }\n      subtitle\n      title\n    }\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  fragment ArticleMarkdownFragment on Article {\n    markdown\n  }\n"): (typeof documents)["\n  fragment ArticleMarkdownFragment on Article {\n    markdown\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  fragment ArticlePreviewFragment on Article {\n    author {\n      name\n    }\n    createdAt\n    hero {\n      caption\n      title\n      subtitle\n    }\n    id\n    slug\n  }\n"): (typeof documents)["\n  fragment ArticlePreviewFragment on Article {\n    author {\n      name\n    }\n    createdAt\n    hero {\n      caption\n      title\n      subtitle\n    }\n    id\n    slug\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  fragment PageContentQueryFragment on Query {\n    page(where: { slug: $slug }) {\n      ...PageHeroFragment\n      ...PageSectionExplorerFragment\n    }\n  }\n"): (typeof documents)["\n  fragment PageContentQueryFragment on Query {\n    page(where: { slug: $slug }) {\n      ...PageHeroFragment\n      ...PageSectionExplorerFragment\n    }\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  fragment PageHeroFragment on Page {\n    hero {\n      caption\n      image {\n        url\n      }\n      subtitle\n      title\n    }\n  }\n"): (typeof documents)["\n  fragment PageHeroFragment on Page {\n    hero {\n      caption\n      image {\n        url\n      }\n      subtitle\n      title\n    }\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  fragment PageSectionExplorerFragment on Page {\n    sections {\n      __typename\n      ...ArticleExplorerSectionFragment\n    }\n  }\n"): (typeof documents)["\n  fragment PageSectionExplorerFragment on Page {\n    sections {\n      __typename\n      ...ArticleExplorerSectionFragment\n    }\n  }\n"];

export function graphql(source: string) {
  return (documents as any)[source] ?? {};
}

export type DocumentType<TDocumentNode extends DocumentNode<any, any>> = TDocumentNode extends DocumentNode<  infer TType,  any>  ? TType  : never;