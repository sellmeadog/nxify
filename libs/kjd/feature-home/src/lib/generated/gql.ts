/* eslint-disable */
import * as types from './graphql';
import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';

/**
 * Map of all GraphQL operations in the project.
 *
 * This map has several performance disadvantages:
 * 1. It is not tree-shakeable, so it will include all operations in the project.
 * 2. It is not minifiable, so the string of a GraphQL query will be multiple times inside the bundle.
 * 3. It does not support dead code elimination, so it will add unused operations.
 *
 * Therefore it is highly recommended to use the babel or swc plugin for production.
 */
const documents = {
    "\n  query HomePage {\n    ...ArticlePreviewList\n  }\n": types.HomePageDocument,
    "fragment ArticlePreview on Article {\n  author {\n    id\n    name\n  }\n  createdAt\n  excerpt\n  id\n  slug\n  title\n}": types.ArticlePreviewFragmentDoc,
    "fragment ArticlePreviewList on Query {\n  articles(first: 10, orderBy: createdAt_DESC) {\n    __typename\n    ...ArticlePreview\n  }\n}": types.ArticlePreviewListFragmentDoc,
};

/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 *
 *
 * @example
 * ```ts
 * const query = graphql(`query GetUser($id: ID!) { user(id: $id) { name } }`);
 * ```
 *
 * The query argument is unknown!
 * Please regenerate the types.
 */
export function graphql(source: string): unknown;

/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  query HomePage {\n    ...ArticlePreviewList\n  }\n"): (typeof documents)["\n  query HomePage {\n    ...ArticlePreviewList\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "fragment ArticlePreview on Article {\n  author {\n    id\n    name\n  }\n  createdAt\n  excerpt\n  id\n  slug\n  title\n}"): (typeof documents)["fragment ArticlePreview on Article {\n  author {\n    id\n    name\n  }\n  createdAt\n  excerpt\n  id\n  slug\n  title\n}"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "fragment ArticlePreviewList on Query {\n  articles(first: 10, orderBy: createdAt_DESC) {\n    __typename\n    ...ArticlePreview\n  }\n}"): (typeof documents)["fragment ArticlePreviewList on Query {\n  articles(first: 10, orderBy: createdAt_DESC) {\n    __typename\n    ...ArticlePreview\n  }\n}"];

export function graphql(source: string) {
  return (documents as any)[source] ?? {};
}

export type DocumentType<TDocumentNode extends DocumentNode<any, any>> = TDocumentNode extends DocumentNode<  infer TType,  any>  ? TType  : never;