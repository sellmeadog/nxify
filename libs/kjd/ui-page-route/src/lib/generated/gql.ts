/* eslint-disable */
import * as types from './graphql';
import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';

/**
 * Map of all GraphQL operations in the project.
 *
 * This map has several performance disadvantages:
 * 1. It is not tree-shakeable, so it will include all operations in the project.
 * 2. It is not minifiable, so the string of a GraphQL query will be multiple times inside the bundle.
 * 3. It does not support dead code elimination, so it will add unused operations.
 *
 * Therefore it is highly recommended to use the babel or swc plugin for production.
 */
const documents = {
    "\n  fragment ArticleExplorerContent on ArticleExplorer {\n    first\n    id\n    subtitle\n    title\n  }\n": types.ArticleExplorerContentFragmentDoc,
    "\n  query RecentArticles($first: Int = 10) {\n    articles: articlesConnection(first: $first, orderBy: createdAt_DESC) {\n      edges {\n        node {\n          createdAt\n          hero {\n            caption\n            title\n          }\n          id\n          slug\n        }\n      }\n      pageInfo {\n        endCursor\n        hasNextPage\n        hasPreviousPage\n        pageSize\n        startCursor\n      }\n    }\n  }\n": types.RecentArticlesDocument,
    "\n  fragment HeroSectionContent on HeroSection {\n    caption\n    id\n    image {\n      url\n    }\n    subtitle\n    title\n  }\n": types.HeroSectionContentFragmentDoc,
    "\n  query PageBySlug($slug: String = \"\") {\n    page(where: { slug: $slug }) {\n      hero {\n        ...HeroSectionContent\n      }\n      sections {\n        __typename\n        ... on ArticleExplorer {\n          ...ArticleExplorerContent\n        }\n        ... on TagExplorer {\n          id\n          first\n          subtitle\n          title\n        }\n      }\n    }\n  }\n": types.PageBySlugDocument,
};

/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 *
 *
 * @example
 * ```ts
 * const query = graphql(`query GetUser($id: ID!) { user(id: $id) { name } }`);
 * ```
 *
 * The query argument is unknown!
 * Please regenerate the types.
 */
export function graphql(source: string): unknown;

/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  fragment ArticleExplorerContent on ArticleExplorer {\n    first\n    id\n    subtitle\n    title\n  }\n"): (typeof documents)["\n  fragment ArticleExplorerContent on ArticleExplorer {\n    first\n    id\n    subtitle\n    title\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  query RecentArticles($first: Int = 10) {\n    articles: articlesConnection(first: $first, orderBy: createdAt_DESC) {\n      edges {\n        node {\n          createdAt\n          hero {\n            caption\n            title\n          }\n          id\n          slug\n        }\n      }\n      pageInfo {\n        endCursor\n        hasNextPage\n        hasPreviousPage\n        pageSize\n        startCursor\n      }\n    }\n  }\n"): (typeof documents)["\n  query RecentArticles($first: Int = 10) {\n    articles: articlesConnection(first: $first, orderBy: createdAt_DESC) {\n      edges {\n        node {\n          createdAt\n          hero {\n            caption\n            title\n          }\n          id\n          slug\n        }\n      }\n      pageInfo {\n        endCursor\n        hasNextPage\n        hasPreviousPage\n        pageSize\n        startCursor\n      }\n    }\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  fragment HeroSectionContent on HeroSection {\n    caption\n    id\n    image {\n      url\n    }\n    subtitle\n    title\n  }\n"): (typeof documents)["\n  fragment HeroSectionContent on HeroSection {\n    caption\n    id\n    image {\n      url\n    }\n    subtitle\n    title\n  }\n"];
/**
 * The graphql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function graphql(source: "\n  query PageBySlug($slug: String = \"\") {\n    page(where: { slug: $slug }) {\n      hero {\n        ...HeroSectionContent\n      }\n      sections {\n        __typename\n        ... on ArticleExplorer {\n          ...ArticleExplorerContent\n        }\n        ... on TagExplorer {\n          id\n          first\n          subtitle\n          title\n        }\n      }\n    }\n  }\n"): (typeof documents)["\n  query PageBySlug($slug: String = \"\") {\n    page(where: { slug: $slug }) {\n      hero {\n        ...HeroSectionContent\n      }\n      sections {\n        __typename\n        ... on ArticleExplorer {\n          ...ArticleExplorerContent\n        }\n        ... on TagExplorer {\n          id\n          first\n          subtitle\n          title\n        }\n      }\n    }\n  }\n"];

export function graphql(source: string) {
  return (documents as any)[source] ?? {};
}

export type DocumentType<TDocumentNode extends DocumentNode<any, any>> = TDocumentNode extends DocumentNode<  infer TType,  any>  ? TType  : never;