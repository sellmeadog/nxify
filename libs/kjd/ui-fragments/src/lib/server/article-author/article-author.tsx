import { SectionLayout } from '@nxify/kjd-ui-layout';
import { FragmentType, unmaskFragment, graphql } from '../../generated';

export const ArticleAuthorFragment = graphql(`
  fragment ArticleAuthorFragment on Article {
    author {
      name
      avatar {
        url
      }
      biography
    }
  }
`);

export interface ArticleAuthorProps {
  data: FragmentType<typeof ArticleAuthorFragment>;
}

export function ArticleAuthor({ data }: ArticleAuthorProps) {
  const { author } = unmaskFragment(ArticleAuthorFragment, data);

  return (
    <SectionLayout className="flex flex-col md:flex-row gap-8 items-center md:items-start prose-headings:mt-0 prose-img:m-0">
      <img
        alt={author?.name}
        className="bg-neutral-950 rounded-full w-1/3 md:w-1/4 aspect-square"
        src={author?.avatar?.url}
      />
      <section>
        <h3>Hi, I'm {author?.name}</h3>
        <p>{author?.biography}</p>
      </section>
    </SectionLayout>
  );
}
