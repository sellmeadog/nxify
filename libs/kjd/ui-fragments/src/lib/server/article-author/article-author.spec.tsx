import React from 'react';
import { render } from '@testing-library/react';

import { ArticleAuthor, ArticleAuthorFragment } from './article-author';
import { makeFragmentData } from '../../generated';

describe('ArticleAuthor', () => {
  it('should render successfully', () => {
    const fragment = makeFragmentData(
      {
        author: {
          name: 'Kennie Davis',
          avatar: {
            url: 'https://media.graphassets.com/mIH7lhHRTEiQU18HLARU',
          },
          biography:
            "An Arizona native, father of three, and software engineer passionate about Nx, React, GraphQL and improving both the developer and user experience. I write about what I'm working on and learning along the way.",
          socialIcons: [
            {
              network: 'github',
              href: 'https://github.com/sellmeadog/nxify',
            },
            {
              network: 'linkedin',
              href: 'https://linkedin.com/in/kenniejaydavis',
            },
            {
              network: 'x',
              href: 'https://twitter.com/kenniejaydavis',
            },
          ],
        },
      },
      ArticleAuthorFragment
    );

    const { baseElement } = render(<ArticleAuthor data={fragment} />);
    expect(baseElement).toMatchSnapshot();
  });
});
