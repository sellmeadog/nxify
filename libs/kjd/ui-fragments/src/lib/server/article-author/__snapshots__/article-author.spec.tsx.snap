// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleAuthor should render successfully 1`] = `
<body>
  <div>
    <section
      class="odd:bg-neutral-700 max-w-none"
    >
      <div
        class="max-w-prose mx-auto p-8 md:px-0 lg:py-16 flex flex-col md:flex-row gap-8 items-center md:items-start prose-headings:mt-0 prose-img:m-0"
      >
        <img
          alt="Kennie Davis"
          class="bg-neutral-950 rounded-full w-1/3 md:w-1/4 aspect-square"
          src="https://media.graphassets.com/mIH7lhHRTEiQU18HLARU"
        />
        <section>
          <h3>
            Hi, I'm 
            Kennie Davis
          </h3>
          <p>
            An Arizona native, father of three, and software engineer passionate about Nx, React, GraphQL and improving both the developer and user experience. I write about what I'm working on and learning along the way.
          </p>
          <div
            class="flex flex-row justify-evenly md:justify-start md:gap-x-2"
          >
            <a
              aria-label="github"
              class="social-icon"
              href="https://github.com/sellmeadog/nxify"
              style="display: inline-block; width: 30px; height: 30px; position: relative; overflow: hidden; vertical-align: middle;"
            >
              <div
                class="social-container"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
              >
                <svg
                  aria-label="github social icon"
                  class="social-svg"
                  role="img"
                  style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; border-radius: 50%; fill-rule: evenodd;"
                  viewBox="0 0 64 64"
                >
                  <g
                    class="social-svg-icon"
                    style="transition: fill 170ms ease-in-out; fill: white;"
                  >
                    <path
                      d="M0,0H64V64H0ZM0,0v64h64V0H0z M37.1,47.2c-0.8,0.2-1.1-0.3-1.1-0.8c0-0.5,0-2.3,0-4.4c0-1.5-0.5-2.5-1.1-3 c3.6-0.4,7.3-1.7,7.3-7.9c0-1.7-0.6-3.2-1.6-4.3c0.2-0.4,0.7-2-0.2-4.2c0,0-1.3-0.4-4.4,1.6c-1.3-0.4-2.6-0.5-4-0.5 c-1.4,0-2.7,0.2-4,0.5c-3.1-2.1-4.4-1.6-4.4-1.6c-0.9,2.2-0.3,3.8-0.2,4.2c-1,1.1-1.6,2.5-1.6,4.3c0,6.1,3.7,7.5,7.3,7.9 c-0.5,0.4-0.9,1.1-1,2.1c-0.9,0.4-3.2,1.1-4.7-1.3c0,0-0.8-1.5-2.5-1.6c0,0-1.6,0-0.1,1c0,0,1,0.5,1.8,2.3c0,0,0.9,3.1,5.4,2.1 c0,1.3,0,2.3,0,2.7c0,0.4-0.3,0.9-1.1,0.8C20.6,45.1,16,39.1,16,32c0-8.8,7.2-16,16-16c8.8,0,16,7.2,16,16 C48,39.1,43.4,45.1,37.1,47.2z"
                    />
                  </g>
                  <g
                    class="social-svg-mask"
                    style="transition: fill 170ms ease-in-out; fill: rgb(10 10 10);"
                  >
                    <path
                      d="M0,0v64h64V0H0z M37.1,47.2c-0.8,0.2-1.1-0.3-1.1-0.8c0-0.5,0-2.3,0-4.4c0-1.5-0.5-2.5-1.1-3 c3.6-0.4,7.3-1.7,7.3-7.9c0-1.7-0.6-3.2-1.6-4.3c0.2-0.4,0.7-2-0.2-4.2c0,0-1.3-0.4-4.4,1.6c-1.3-0.4-2.6-0.5-4-0.5 c-1.4,0-2.7,0.2-4,0.5c-3.1-2.1-4.4-1.6-4.4-1.6c-0.9,2.2-0.3,3.8-0.2,4.2c-1,1.1-1.6,2.5-1.6,4.3c0,6.1,3.7,7.5,7.3,7.9 c-0.5,0.4-0.9,1.1-1,2.1c-0.9,0.4-3.2,1.1-4.7-1.3c0,0-0.8-1.5-2.5-1.6c0,0-1.6,0-0.1,1c0,0,1,0.5,1.8,2.3c0,0,0.9,3.1,5.4,2.1 c0,1.3,0,2.3,0,2.7c0,0.4-0.3,0.9-1.1,0.8C20.6,45.1,16,39.1,16,32c0-8.8,7.2-16,16-16c8.8,0,16,7.2,16,16 C48,39.1,43.4,45.1,37.1,47.2z"
                    />
                  </g>
                </svg>
              </div>
            </a>
            <a
              aria-label="linkedin"
              class="social-icon"
              href="https://linkedin.com/in/kenniejaydavis"
              style="display: inline-block; width: 30px; height: 30px; position: relative; overflow: hidden; vertical-align: middle;"
            >
              <div
                class="social-container"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
              >
                <svg
                  aria-label="linkedin social icon"
                  class="social-svg"
                  role="img"
                  style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; border-radius: 50%; fill-rule: evenodd;"
                  viewBox="0 0 64 64"
                >
                  <g
                    class="social-svg-icon"
                    style="transition: fill 170ms ease-in-out; fill: white;"
                  >
                    <path
                      d="M0,0H64V64H0ZM0,0v64h64V0H0z M25.8,44h-5.4V26.6h5.4V44z M23.1,24.3c-1.7,0-3.1-1.4-3.1-3.1c0-1.7,1.4-3.1,3.1-3.1 c1.7,0,3.1,1.4,3.1,3.1C26.2,22.9,24.8,24.3,23.1,24.3z M46,44h-5.4v-8.4c0-2,0-4.6-2.8-4.6c-2.8,0-3.2,2.2-3.2,4.5V44h-5.4V26.6 h5.2V29h0.1c0.7-1.4,2.5-2.8,5.1-2.8c5.5,0,6.5,3.6,6.5,8.3V44z"
                    />
                  </g>
                  <g
                    class="social-svg-mask"
                    style="transition: fill 170ms ease-in-out; fill: rgb(10 10 10);"
                  >
                    <path
                      d="M0,0v64h64V0H0z M25.8,44h-5.4V26.6h5.4V44z M23.1,24.3c-1.7,0-3.1-1.4-3.1-3.1c0-1.7,1.4-3.1,3.1-3.1 c1.7,0,3.1,1.4,3.1,3.1C26.2,22.9,24.8,24.3,23.1,24.3z M46,44h-5.4v-8.4c0-2,0-4.6-2.8-4.6c-2.8,0-3.2,2.2-3.2,4.5V44h-5.4V26.6 h5.2V29h0.1c0.7-1.4,2.5-2.8,5.1-2.8c5.5,0,6.5,3.6,6.5,8.3V44z"
                    />
                  </g>
                </svg>
              </div>
            </a>
            <a
              aria-label="x"
              class="social-icon"
              href="https://twitter.com/kenniejaydavis"
              style="display: inline-block; width: 30px; height: 30px; position: relative; overflow: hidden; vertical-align: middle;"
            >
              <div
                class="social-container"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
              >
                <svg
                  aria-label="x social icon"
                  class="social-svg"
                  role="img"
                  style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; border-radius: 50%; fill-rule: evenodd;"
                  viewBox="0 0 64 64"
                >
                  <g
                    class="social-svg-icon"
                    style="transition: fill 170ms ease-in-out; fill: white;"
                  >
                    <path
                      d="M0,0H64V64H0ZM 0 0 L 0 64 L 64 64 L 64 0 L 0 0 z M 16 17.537109 L 26.125 17.537109 L 33.117188 26.779297 L 41.201172 17.537109 L 46.109375 17.537109 L 35.388672 29.789062 L 48 46.462891 L 38.125 46.462891 L 30.390625 36.351562 L 21.541016 46.462891 L 16.632812 46.462891 L 28.097656 33.357422 L 16 17.537109 z M 21.730469 20.320312 L 39.480469 43.525391 L 42.199219 43.525391 L 24.648438 20.320312 L 21.730469 20.320312 z "
                    />
                  </g>
                  <g
                    class="social-svg-mask"
                    style="transition: fill 170ms ease-in-out; fill: rgb(10 10 10);"
                  >
                    <path
                      d="M 0 0 L 0 64 L 64 64 L 64 0 L 0 0 z M 16 17.537109 L 26.125 17.537109 L 33.117188 26.779297 L 41.201172 17.537109 L 46.109375 17.537109 L 35.388672 29.789062 L 48 46.462891 L 38.125 46.462891 L 30.390625 36.351562 L 21.541016 46.462891 L 16.632812 46.462891 L 28.097656 33.357422 L 16 17.537109 z M 21.730469 20.320312 L 39.480469 43.525391 L 42.199219 43.525391 L 24.648438 20.320312 L 21.730469 20.320312 z "
                    />
                  </g>
                </svg>
              </div>
            </a>
          </div>
        </section>
      </div>
    </section>
  </div>
</body>
`;
